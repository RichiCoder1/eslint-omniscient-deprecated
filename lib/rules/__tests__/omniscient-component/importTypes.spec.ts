import { fix, withRuleOptions } from "./__setup__/setup";

describe("import types", () => {
    const input = `
import component from 'omniscient';

const TestComponent = component(() => {
    return (<h1>Test</h1>);
})`;

    it("should handle existing named import", () => {
        const customInput = `import { Component } from 'react';${input}`;
        const report = fix(customInput);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should handle existing default import", () => {
        const customInput = `import React from 'react';${input}`;
        const report = fix(customInput);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should handle default import types", () => {
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should handle custom import types", () => {
        const report = fix(
            input,
            withRuleOptions({
                componentImport: "MyComponent",
                componentModule: "MyLib",
            })
        );
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should handle already present custom import types", () => {
        const customInput = `import MyLib from 'MyLib';${input}`;
        const report = fix(
            customInput,
            withRuleOptions({
                componentImport: "MyComponent",
                componentModule: "MyLib",
            })
        );
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should handle double import fixes", () => {
        const report = fix(
            input,
            withRuleOptions({
                memoImport: "memo",
                memoModule: "MyLib",
                areEqualImport: "areEqual",
                areEqualModule: "MyLib",
                passAreEqualToMemo: true,
            })
        );
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should handle double import fixes on already present type", () => {
        const customInput = `import MyLib from 'MyLib';${input}`;
        const report = fix(
            customInput,
            withRuleOptions({
                memoImport: "memo",
                memoModule: "MyLib",
                areEqualImport: "areEqual",
                areEqualModule: "MyLib",
                passAreEqualToMemo: true,
            })
        );
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });
});
