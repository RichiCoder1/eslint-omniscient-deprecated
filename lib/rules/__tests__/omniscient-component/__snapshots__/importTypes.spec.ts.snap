// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import types should handle already present custom import types 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { isEqualImmutable } from 'ImmutableComponent';
import MyLib from 'MyLib';
import component from 'omniscient';

class TestComponent extends MyLib.MyComponent {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        return <h1>Test</h1>;
    }
}",
}
`;

exports[`import types should handle custom import types 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { MyComponent } from 'MyLib';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends MyComponent {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        return <h1>Test</h1>;
    }
}",
}
`;

exports[`import types should handle default import types 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        return <h1>Test</h1>;
    }
}",
}
`;

exports[`import types should handle double import fixes 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { memo, areEqual } from 'MyLib';

import component from 'omniscient';

const TestComponent = memo(function TestComponent() {
    return <h1>Test</h1>;
}, areEqual);",
}
`;

exports[`import types should handle double import fixes on already present type 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import MyLib from 'MyLib';
import component from 'omniscient';

const TestComponent = MyLib.memo(function TestComponent() {
    return <h1>Test</h1>;
}, MyLib.areEqual);",
}
`;

exports[`import types should handle existing default import 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { isEqualImmutable } from 'ImmutableComponent';
import React from 'react';
import component from 'omniscient';

class TestComponent extends React.Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        return <h1>Test</h1>;
    }
}",
}
`;

exports[`import types should handle existing named import 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { isEqualImmutable } from 'ImmutableComponent';
import { Component } from 'react';
import component from 'omniscient';

class TestComponent extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        return <h1>Test</h1>;
    }
}",
}
`;
